fhir-mapping/src/main/ml-modules/root/data-services/utils/ctsQueryUtils.sjs

```diff
@@ -0,0 +1,25 @@
+const egress = require('../../fhir-accelerator/egress-mapping.sjs')
+
+module.exports = {
+  defaultTextSearchModifiers: ['case-insensitive', 'wildcarded', 'whitespace-insensitive', 'punctuation-insensitive'],
+
+  getValueAndSystemFromString(str) {
+    const idx = str.lastIndexOf('|');
+    const system = idx === -1 ? '' : str.slice(0, idx);
+    const value = str.slice(idx + 1);
+
+    return [value, system];
+  },
+
+  searchToQuery(fieldToQueryMap) {
+    return ({ field, modifier, values }) => {
+      if (field in fieldToQueryMap) {
+        return fieldToQueryMap[field](values, modifier);
+      }
+
+      const searchValues = egress.searchValuesWithModifier(values, modifier);
```

Comment:
The application of the modifiers should only happen to string parameters. this transform should probably happen in the field specific functions for the ones that are relevant.